File: MyVinted.Client/src/app/models/domain/message.ts
export interface Message {
  id: string;
  text: string;
  dateSent: Date;
  senderId: string;
  recipientId: string;
  isRead: boolean;
  isLiked: boolean;
  senderName: string;
  senderAvatarUrl: string;
  recipientName: string;
  recipientAvatarUrl: string;
}

File: MyVinted.Core.Domain/Entities/Message.cs
using System;
using MyVinted.Core.Common.Helpers;

namespace MyVinted.Core.Domain.Entities
{
    public class Message
    {
        public string Id { get; protected set; } = Utils.Id();
        public string Text { get; protected set; }
        public DateTime DateSent { get; protected set; } = DateTime.Now;
        public string SenderId { get; protected set; }
        public string RecipientId { get; protected set; }
        public bool IsRead { get; protected set; }
        public bool IsLiked { get; protected set; }

        public virtual User Sender { get; protected set; }
        public virtual User Recipient { get; protected set; }

        public static Message Create(string text) => new Message { Text = text };

        public void MarkAsRead() => IsRead = true;

        public void ToggleIsLiked() => IsLiked = !IsLiked;
    }
}